state/machine_linklayerdevices.go

```diff
@@ -859,9 +859,86 @@ func (m *Machine) AllNetworkAddresses() ([]network.Address, error) {
 	for i := range stateAddresses {
 		networkAddresses[i] = stateAddresses[i].NetworkAddress()
 	}
+	// TODO(jam): 20161130 NetworkAddress object has a SpaceName attribute.
+	// However, we are not filling in that information here.
 	return networkAddresses, nil
 }
 
+
+// Should this be NetworkInterface rather than NetworkInfo?
+type NetworkInfo struct {
+	// If the machine is in the given space, it should have a primary host
+	// device that is in that space, which should be bridged.
+	HostDevice *LinkLayerDevice
+	BridgeDevice *LinkLayerDevice
+}
+
+// BridgeDevicesForSpaces takes a list of spaces, and returns the associated
+// bridge that should be used for that space. If no bridge has been created on
```

Comment:
So we take a list of spaces, and return a map. We could return a list, but if we did, I'd rather put the space name into the object, so that you always know the context of each object. Having not used it in anger, I wasn't sure what would be best. Also, we should consider that we might change it to return lists of bridges/host devices for each space name, since we may end up with more than one host device in the same space.